### PR TITLE
implemented multiple blended styles

### PR DESCRIPTION
**Pull request by kathrynsiegel**

### CODE REVIEW

File: neural_style.py [hunk 1]
```diff
@@ -76,21 +84,31 @@ def main():
...
-    style_image = scipy.misc.imresize(style_image, options.style_scale *
-            target_shape[1] / style_image.shape[1])
+    for i in range(0,num_styles):
+        style_scale = STYLE_SCALE
+        if options.style_scales != None:
+            style_scale = options.style_scales[i]
+        style_images[i] = scipy.misc.imresize(style_images[i], style_scale *
+                target_shape[1] / style_images[i].shape[1])
+
+    style_blend_weights = options.style_blend_weights
```
Comment (anishathalye) [inline neural_style.py]: Should these be normalized?

File: stylize.py [hunk 2]
```diff
@@ -8,14 +8,18 @@
...
-def stylize(network, initial, content, style, iterations,
-        content_weight, style_weight, tv_weight,
+def stylize(network, initial, content, styles, iterations,
+        content_weight, style_weight, style_blend_weights, tv_weight,
         learning_rate, print_iterations=None, checkpoint_iterations=None):
     shape = (1,) + content.shape
-    style_shape = (1,) + style.shape
+    style_shapes = []
+    num_styles = len(styles)
+    for i in range(0,num_styles):
```
Comment (anishathalye) [inline stylize.py]: Don't need the lower bound for the `range()` constructor.
Also, it's more Pythonic to iterate over the objects:

```python
for style in styles:
    style_shapes.append((1,) + style.shape)
```

Or even more concise:

```python
style_shapes = [(1,) + style.shape for style in styles]
```

File: stylize.py [hunk 3]
```diff
@@ -8,14 +8,18 @@
...
+        style_shapes.append((1,) + styles[i].shape)
     content_features = {}
-    style_features = {}
+    style_features = []
+    for i in range(0,num_styles):
+        style_features.append({})
```
Comment (anishathalye) [inline stylize.py]:
```python
style_features = [{} for _ in styles]
```

File: stylize.py [hunk 4]
```diff
@@ -29,14 +33,15 @@ def stylize(network, initial, content, style, iterations,
...
     with g.as_default(), g.device('/cpu:0'), tf.Session() as sess:
-        image = tf.placeholder('float', shape=style_shape)
-        net, _ = vgg.net(network, image)
-        style_pre = np.array([vgg.preprocess(style, mean_pixel)])
-        for layer in STYLE_LAYERS:
-            features = net[layer].eval(feed_dict={image: style_pre})
-            features = np.reshape(features, (-1, features.shape[3]))
-            gram = np.matmul(features.T, features) / features.size
-            style_features[layer] = gram
+        for i in range(0,num_styles):
```
Comment (anishathalye) [inline stylize.py]:
```python
for style in styles:
```

File: stylize.py [hunk 5]
```diff
@@ -54,17 +59,18 @@ def stylize(network, initial, content, style, iterations,
...
-            _, height, width, number = map(lambda i: i.value, layer.get_shape())
-            size = height * width * number
-            feats = tf.reshape(layer, (-1, number))
-            gram = tf.matmul(tf.transpose(feats), feats) / size
-            style_gram = style_features[style_layer]
-            style_losses.append(2 * tf.nn.l2_loss(gram - style_gram) /
-                    style_gram.size)
-        style_loss = style_weight * reduce(tf.add, style_losses)
+        style_loss = 0
+        for i in range(0,num_styles):
```
Comment (anishathalye) [inline stylize.py]: `range(len(styles))` (don't need lower bound)

File: stylize.py (hunk 4 above)
Comment (anishathalye) [inline stylize.py]: `range(len(styles))` (don't need lower bound)

**Comment by anishathalye** [comment]
Nice!

If you have a good example, wanna add it to the readme?

File: neural_style.py (hunk 1 above)
Comment (kathrynsiegel) [inline neural_style.py]: hmm..I was envisioning them passed in as normalized already, but I'll fix this so it auto-normalizes

**Comment by anishathalye** [comment]
Merged 1036f0502558e3d7595bd886c80023cb9ed7f405 in bfdbc075d0be3ebc3acd82bebe81e3831273dd09.
